Templates/CSharp/Requests/MethodRequestBuilder.cs.tt

```diff
@@ -10,7 +10,7 @@ var requestType = entityName + methodName + "Request";
 var requestBuilderType = requestType + "Builder";
 
 var isPost = method.IsAction() && method.Parameters != null && method.Parameters.Any();
-var methodType = isPost ? "Post" : "Get";
+var methodType = method.IsFunction ? "Function" : "Action";
```

Comment:
The existing code assumed that OData actions were always POST, and OData functions were always GET. We have composable OData functions that support other verbs. 

Since OData functions can be used with GET, PUT, PATCH, I've made changes here and in other places to support the PUT and PATCH scenarios for functions.